### PR TITLE
*: move show tests.

### PR DESCRIPTION
**Pull request by hanfei1991**
1. move test of show @shenli 
2. move test of Database @shenli 
3. move test of ExecRestrictedSQL @shenli 
4. remove test sleep, it's tested in expression package @zimulala 
5. remove index point look up @shenli

### CODE REVIEW

File: session_test.go [hunk 1]
```diff
@@ -792,69 +792,6 @@ func (s *testSessionSuite) TestMySQLTypes(c *C) {
...
-	c.Assert(row.Data[0].GetString(), Equals, "t")
-
-	r = mustExecSQL(c, se, fmt.Sprintf("show table status from %s like 't';", dbName))
-	row, err = r.Next()
-	c.Assert(err, IsNil)
-	c.Assert(row.Data, HasLen, 18)
-	c.Assert(row.Data[0].GetString(), Equals, "t")
-	c.Assert(row.Data[17].GetString(), Equals, "注释")
-
-	r = mustExecSQL(c, se, "show databases like 'test'")
```
Comment (coocood) [inline session_test.go]: Why remove this?

File: executor/show_test.go [hunk 2]
```diff
@@ -361,3 +361,33 @@ func (s *testSuite) TestIssue3641(c *C) {
...
 	c.Assert(err.Error(), Equals, plan.ErrNoDB.Error())
 }
+
+// TestShow2 is moved from session_test
+func (s *testSuite) TestShow2(c *C) {
+	tk := testkit.NewTestKit(c, s.store)
+	tk.MustExec("use test")
+
+	tk.MustExec("set global autocommit=1")
+	tk.MustQuery("show global variables where variable_name = 'autocommit'").Check(testkit.Rows("autocommit 1"))
```
Comment (coocood) [inline executor/show_test.go]: Why did it change from `ON` to `1`?

Comment (hanfei1991) [reply executor/show_test.go]: This is changed in other test, I have fixed it.

File: session_test.go (hunk 1 above)
Comment (hanfei1991) [reply session_test.go]: fixed

**Comment by hanfei1991** [comment]
@coocood PTAL

**Comment by coocood** [comment]
@hanfei1991 
I tested this in MySQL, no matter set `1` or `ON`, the show result is always `ON`.

**Comment by hanfei1991** [comment]
@coocood But TiDB will show the original value... we can create an issue.

**Comment by coocood** [comment]
@hanfei1991 
Since this is just a small issue, add a TODO will be fine.

**Comment by hanfei1991** [comment]
@coocood PTAL

**Comment by coocood** [comment]
LGTM

**Pull request merged**